src/scenes/ProgressReports/EditForm/ProgressReportContext.tsx

```diff
@@ -53,11 +58,29 @@ export const ProgressReportContextProvider = ({
   const incompleteSteps = useMemo(() => {
     return Object.fromEntries(
       Object.entries(groupedStepMap).flatMap(([title, steps]) => {
-        const incompleteSteps = steps.filter(([_, { isIncomplete }]) =>
-          report == null
-            ? true
-            : isIncomplete?.({ report, currentUserRoles }) ?? false
-        );
+        const incompleteSteps = steps
+          .map(([label, step]) => {
+            const incompleteStatus = report
+              ? step.isIncomplete?.({ report, currentUserRoles }) ?? {
+                  isIncomplete: false,
+                  severity: 'suggested',
+                }
+              : { isIncomplete: true, severity: 'suggested' };
+            return { label, step, incompleteStatus };
+          })
+          .filter((item) => item.incompleteStatus.isIncomplete)
+          .map((item) => {
+            const {
+              label,
+              step,
+              incompleteStatus: { severity },
+            } = item;
+            return [label, step, severity] as [
+              string,
+              StepComponent,
+              NotificationSeverityType
+            ];
+          });
```

Comment:
I would combine the map & filter into flatMap to simplify and to not have weird fallback values.
Also there shouldn't be a need to type cast the final tuple - that can hide errors. If actually needed for some reason, which I'm not convinced it is, you should you use `satisfies` instead of `as`.

```ts
const incompleteSteps = steps
  .flatMap(([label, step]) => {
    if (!report || !step.isIncomplete) {
      return [];
    }
    const { isIncomplete, severity } = step
      .isIncomplete({ report, currentUserRoles });
    return [[label, step, severity] as const];
  });
```

Reply:
@CarsonF I made an attempt at the revisions inside the ProgressReportContext file, and although it looks like the incompleteSteps function starting on line 57 is outputting the correct shape, I'm getting an error with the shape down on line 93.  The error I get is:

`Type '{ [k: string]: (readonly [string, StepComponent, IncompleteSeverity])[]; }' is missing the following properties from type 'readonly [label: string, step: StepComponent, severity: IncompleteSeverity][]': length, concat, join, slice, and 26 more.ts(2740)
ProgressReportContext.tsx(15, 3): The expected type comes from property 'incompleteSteps' which is declared here on type 'ProgressReportContext'
(property) ProgressReportContext.incompleteSteps: GroupedIncompleteSteps`

I've tried a bunch of things and tried some chatGPT suggestions, but none seem to resolve my issue. I will need your help with this one.

Reply:
@CarsonF I pushed up another commit trying to conform the code to what you were requesting but had to make other modifications to the type to make it work.  Please take a look, thank you.

Reply:
Ok, I was under the impression that you wanted the type to be literally just the array based on your other feedback, see below.  Sorry, my misunderstanding, thank you.

`Alternatively, you could use an object instead of a tuple here. That might be clearer everywhere.
ReadonlyArray<{
  label: string;
  step: StepComponent;
  severity: IncompleteSeverity;
}>`

Reply:
Yeah sorry I was just referencing the tuple (within record) to change to an object (within the record).